subprojects/logging/src/main/java/org/gradle/internal/logging/serializer/ProgressStartEventSerializer.java

```diff
@@ -25,31 +25,38 @@
 public class ProgressStartEventSerializer implements Serializer<ProgressStartEvent> {
     @Override
     public void write(Encoder encoder, ProgressStartEvent event) throws Exception {
-        encoder.writeSmallLong(event.getOperationId().getId());
-        if (event.getParentId() == null) {
+        encoder.writeSmallLong(event.getProgressOperationId().getId());
+        if (event.getParentProgressOperationId() == null) {
             encoder.writeBoolean(false);
         } else {
             encoder.writeBoolean(true);
-            encoder.writeSmallLong(event.getParentId().getId());
+            encoder.writeSmallLong(event.getParentProgressOperationId().getId());
```

Comment:
Could small long become an issue for bigger builds e.g. 500 subproject that produce a lot of operation IDs?